---
title: "How to use biopathscore"
author: "Sen liang, Tianci Song, Wei Du"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

The aberrant alterations of biological functions are well known in tumorigenesis and cancer development. Hence, with advances in high-throughput sequencing technologies, capturing and quantifying the functional alterations in cancers based on expression profiles to explore cancer malignant process is highlighted as one of the important topics among cancer researches. 

Now, we propose an algorithm for quantifying biological processes by using gene expression profiles over a sample population, which involves the idea of constructing principal curves to condense information of each biological process by a novel scoring scheme on an individualized manner. And this article was published at Scientific Reports[Paper #SREP-17-11363B], "A Novel Unsupervised Algorithm for Biological Process-based Analysis on Cancer" by Tianci song, Sha Cao, Tao Sheng, Sen Liang, Wei Du, and Yanchun Liang.

Upon requests from users and collaborators, we developed this R package of "biopathscore".

## How to cite
```{r, warning=FALSE}
citation("biopathscore")
```


## Install and Load

### install from Github

```{r, eval=FALSE}
install.packages("devtools")
devtools::install_github("liangsen35/biopathscore")
library("biopathscore")

```
### install from CRAN (!!!!not availiable now!!!)
```{r, eval=FALSE}
# install.packages("biopathscore")
```



## Internal Data
- `dat_brca` a matrix, the TCGA-BRCA cancer gene expression data, each columu is a sample, each row is a gene(present by Gene ID), 
- `v_brca_normals` a vector, the dat_brca's samples, if it is Normal then TRUE, else is Tumor then FALSE.
- `l_kegg_gs` a list, each list is a gene ID vector of a KEGG pathway, total 186 KEGG pathway.
- `l_kegg_gs_min` a small list of l_kegg_gs for test.

```{r, results='hide'}
library(biopathscore)
data("dat_brca")
data('v_brca_normals')
data('l_kegg_gs_min')

dat_brca[1:10, 1:10]
```

```{r, echo=FALSE, results='asis'}
knitr::kable(dat_brca[1:10, 1:10])
```

```{r}
dim(dat_brca)
length(v_brca_normals)
head(v_brca_normals)
length(l_kegg_gs_min)
head(l_kegg_gs_min$KEGG_P53_SIGNALING_PATHWAY)

```

## Functions
- `get_biopath_score()` run the algorithm and get the bio-pathway score for each sample 
- `plot3D_lpc()` plot the 3D graphics of the result

### (1) get_biopath_score 
```r
get_biopath_score = function(data, biopath_genes,normals=NULL, ranks=NULL,attempts = 50, maximize_stability=F, use_min = T, min_exp = 2, min_std = 0.1)
```

> more information, see ?get_biopath_score


- data. a matrix, row represent gene, col represent sample, must contain row names(Gene ID) and col names(Sample ID)
- biopath_genes.  a vector, the Gene ID of a bio-pathway you want to caculate.
- normals.  a vector, TURE or FALSE of the samples, if sample is Normal then TRUE, else is Tumor then FALSE
- ranks.  a vector or NULL, if not have normals, rank your samples
- attempts. a non-negative integer, default 50, iter times
- maximize_stability. maximize_stability=T, then algorithm will run more times to maximize score stability, very slow.
- min_exp.  use min_exp replace gene expression which lower than it, if min_exp = NULL, than not replace, default 2
- min_std.  use min std to cut-off gene whose std below than it, min_std must > 0, default 0.1



### (2) plot3D_lpc 
```r
plot3D_lpc = function(res.lpc, normals, drawLine = F, outputHTML = F, outputDir = "./plot", fileName = "plot")
```

> more information, see ?plot3D_lpc

- res.lpc.  data after run get_biopath_score
- normals.  a vector of the samples, if samples is Normal then TRUE, else Tumor then FALSE
- outputHTML. if need output this plot as a html file is TRUE, if not then FALSE
- outputDir.  if outputHTML=T, this is availiable, the output file directory
- fileName. if outputHTML=T, this is availiable, the output file name.



## Examples
```{R, eval = F}
library(biopathscore)
data("dat_brca")
data('v_brca_normals')
data('l_kegg_gs_min')


test_pathway = l_kegg_gs_min$KEGG_DNA_REPLICATION


### Examples 1: not use maximize_stability = F, (if run slow, set attempts lower number)
pds = get_biopath_score(data = dat_brca, biopath_genes = test_pathway, normals = v_brca_normals,
                        attempts = 100, maximize_stability = F)


plot3D_lpc(pds, v_brca_normals,drawLine = F, outputHTML = F )
plot3D_lpc(pds, v_brca_normals,drawLine = T, outputHTML = F )


### Examples 1: use maximize_stability = T, very slow
pds2 = get_biopath_score(data = dat_brca, biopath_genes = test_pathway, normals = v_brca_normals,
                        attempts = 100, maximize_stability = T)


plot3D_lpc(pds2, v_brca_normals,drawLine = F, outputHTML = F )
plot3D_lpc(pds2, v_brca_normals,drawLine = T, outputHTML = F )


```

