<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Sen liang, Tianci Song, Wei Du" />

<meta name="date" content="2017-06-19" />

<title>How to use biopathscore</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">How to use biopathscore</h1>
<h4 class="author"><em>Sen liang, Tianci Song, Wei Du</em></h4>
<h4 class="date"><em>2017-06-19</em></h4>



<p>The aberrant alterations of biological functions are well known in tumorigenesis and cancer development. Hence, with advances in high-throughput sequencing technologies, capturing and quantifying the functional alterations in cancers based on expression profiles to explore cancer malignant process is highlighted as one of the important topics among cancer researches.</p>
<p>Now, we propose an algorithm for quantifying biological processes by using gene expression profiles over a sample population, which involves the idea of constructing principal curves to condense information of each biological process by a novel scoring scheme on an individualized manner. And this article was published at Scientific Reports[Paper #SREP-17-11363B], “A Novel Unsupervised Algorithm for Biological Process-based Analysis on Cancer” by Tianci song, Sha Cao, Tao Sheng, Sen Liang, Wei Du, and Yanchun Liang.</p>
<p>Upon requests from users and collaborators, we developed this R package of “biopathscore”.</p>
<div id="how-to-cite" class="section level2">
<h2>How to cite</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">citation</span>(<span class="st">&quot;biopathscore&quot;</span>)</code></pre></div>
<pre><code>## 
## To cite biopathscore in publications use:
## 
##   Tianci Song(2011). A Novel Unsupervised Algorithm for Biological
##   Process-based Analysis on Cancer. Scientific Reports, 2017. URL
##   http://www.xxxxxxxxxxxxxx.
## 
## A BibTeX entry for LaTeX users is
## 
##   @Article{,
##     title = {A Novel Unsupervised Algorithm for Biological Process-based Analysis on Cancer},
##     author = {Tianci Song and Sha Cao and Sheng Tao and Sen Liang and Wei Du and Yanchun Liang},
##     journal = {Scientific Reports},
##     year = {2017},
##     volume = {xx},
##     number = {xx},
##     pages = {xx},
##     url = {xxxxxxxxxxxx},
##   }</code></pre>
</div>
<div id="install-and-load" class="section level2">
<h2>Install and Load</h2>
<div id="install-from-github" class="section level3">
<h3>install from Github</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)
devtools::<span class="kw">install_github</span>(<span class="st">&quot;liangsen35/biopathscore&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;biopathscore&quot;</span>)</code></pre></div>
</div>
<div id="install-from-cran-not-availiable-now" class="section level3">
<h3>install from CRAN (!!!!not availiable now!!!)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages(&quot;biopathscore&quot;)</span></code></pre></div>
</div>
</div>
<div id="internal-data" class="section level2">
<h2>Internal Data</h2>
<ul>
<li><code>dat_brca</code> a matrix, the TCGA-BRCA cancer gene expression data, each columu is a sample, each row is a gene(present by Gene ID),</li>
<li><code>v_brca_normals</code> a vector, the dat_brca’s samples, if it is Normal then TRUE, else is Tumor then FALSE.</li>
<li><code>l_kegg_gs</code> a list, each list is a gene ID vector of a KEGG pathway, total 186 KEGG pathway.</li>
<li><code>l_kegg_gs_min</code> a small list of l_kegg_gs for test.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(biopathscore)
<span class="kw">data</span>(<span class="st">&quot;dat_brca&quot;</span>)
<span class="kw">data</span>(<span class="st">'v_brca_normals'</span>)
<span class="kw">data</span>(<span class="st">'l_kegg_gs_min'</span>)

dat_brca[<span class="dv">1</span>:<span class="dv">10</span>, <span class="dv">1</span>:<span class="dv">10</span>]</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">TCGA-E2-A15M-N</th>
<th align="right">TCGA-BH-A1FB-N</th>
<th align="right">TCGA-A7-A0DB-N</th>
<th align="right">TCGA-BH-A1FU-N</th>
<th align="right">TCGA-BH-A1FC-N</th>
<th align="right">TCGA-BH-A18P-N</th>
<th align="right">TCGA-A7-A0CH-N</th>
<th align="right">TCGA-BH-A18Q-N</th>
<th align="right">TCGA-BH-A0H9-N</th>
<th align="right">TCGA-BH-A18N-N</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>10</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
</tr>
<tr class="even">
<td>100</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.38</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.17</td>
<td align="right">2.00</td>
</tr>
<tr class="odd">
<td>1000</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
</tr>
<tr class="even">
<td>10000</td>
<td align="right">3.65</td>
<td align="right">3.00</td>
<td align="right">3.16</td>
<td align="right">3.32</td>
<td align="right">2.89</td>
<td align="right">2.93</td>
<td align="right">2.51</td>
<td align="right">3.46</td>
<td align="right">2.56</td>
<td align="right">3.22</td>
</tr>
<tr class="odd">
<td>10001</td>
<td align="right">2.11</td>
<td align="right">2.34</td>
<td align="right">2.00</td>
<td align="right">2.27</td>
<td align="right">2.33</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.44</td>
<td align="right">2.00</td>
<td align="right">2.04</td>
</tr>
<tr class="even">
<td>100049587</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
</tr>
<tr class="odd">
<td>10005</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
<td align="right">2.00</td>
</tr>
<tr class="even">
<td>10006</td>
<td align="right">4.22</td>
<td align="right">4.41</td>
<td align="right">3.94</td>
<td align="right">4.27</td>
<td align="right">4.29</td>
<td align="right">3.85</td>
<td align="right">3.66</td>
<td align="right">4.46</td>
<td align="right">3.43</td>
<td align="right">4.17</td>
</tr>
<tr class="odd">
<td>10007</td>
<td align="right">3.24</td>
<td align="right">3.50</td>
<td align="right">4.14</td>
<td align="right">3.62</td>
<td align="right">3.35</td>
<td align="right">2.77</td>
<td align="right">3.50</td>
<td align="right">3.31</td>
<td align="right">3.07</td>
<td align="right">3.29</td>
</tr>
<tr class="even">
<td>10009</td>
<td align="right">2.97</td>
<td align="right">4.22</td>
<td align="right">2.66</td>
<td align="right">3.64</td>
<td align="right">3.07</td>
<td align="right">3.20</td>
<td align="right">2.53</td>
<td align="right">3.77</td>
<td align="right">2.07</td>
<td align="right">3.40</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(dat_brca)</code></pre></div>
<pre><code>## [1] 12240  1204</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(v_brca_normals)</code></pre></div>
<pre><code>## [1] 1204</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(v_brca_normals)</code></pre></div>
<pre><code>## TCGA-E2-A15M-N TCGA-BH-A1FB-N TCGA-A7-A0DB-N TCGA-BH-A1FU-N TCGA-BH-A1FC-N 
##           TRUE           TRUE           TRUE           TRUE           TRUE 
## TCGA-BH-A18P-N 
##           TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(l_kegg_gs_min)</code></pre></div>
<pre><code>## [1] 5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(l_kegg_gs_min$KEGG_P53_SIGNALING_PATHWAY)</code></pre></div>
<pre><code>##  CASP9  CASP8    SFN   TSC2   IGF1   CDK1 
##  &quot;842&quot;  &quot;841&quot; &quot;2810&quot; &quot;7249&quot; &quot;3479&quot;  &quot;983&quot;</code></pre>
</div>
<div id="functions" class="section level2">
<h2>Functions</h2>
<ul>
<li><code>get_biopath_score()</code> run the algorithm and get the bio-pathway score for each sample</li>
<li><code>plot3D_lpc()</code> plot the 3D graphics of the result</li>
</ul>
<div id="get_biopath_score" class="section level3">
<h3>(1) get_biopath_score</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get_biopath_score =<span class="st"> </span>function(data, biopath_genes,<span class="dt">normals=</span><span class="ot">NULL</span>, <span class="dt">ranks=</span><span class="ot">NULL</span>,<span class="dt">attempts =</span> <span class="dv">50</span>, <span class="dt">maximize_stability=</span>F, <span class="dt">use_min =</span> T, <span class="dt">min_exp =</span> <span class="dv">2</span>, <span class="dt">min_std =</span> <span class="fl">0.1</span>)</code></pre></div>
<blockquote>
<p>more information, see ?get_biopath_score</p>
</blockquote>
<ul>
<li>data. a matrix, row represent gene, col represent sample, must contain row names(Gene ID) and col names(Sample ID)</li>
<li>biopath_genes. a vector, the Gene ID of a bio-pathway you want to caculate.</li>
<li>normals. a vector, TURE or FALSE of the samples, if sample is Normal then TRUE, else is Tumor then FALSE</li>
<li>ranks. a vector or NULL, if not have normals, rank your samples</li>
<li>attempts. a non-negative integer, default 50, iter times</li>
<li>maximize_stability. maximize_stability=T, then algorithm will run more times to maximize score stability, very slow.</li>
<li>min_exp. use min_exp replace gene expression which lower than it, if min_exp = NULL, than not replace, default 2</li>
<li>min_std. use min std to cut-off gene whose std below than it, min_std must &gt; 0, default 0.1</li>
</ul>
</div>
<div id="plot3d_lpc" class="section level3">
<h3>(2) plot3D_lpc</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plot3D_lpc =<span class="st"> </span>function(res.lpc, normals, <span class="dt">drawLine =</span> F, <span class="dt">outputHTML =</span> F, <span class="dt">outputDir =</span> <span class="st">&quot;./plot&quot;</span>, <span class="dt">fileName =</span> <span class="st">&quot;plot&quot;</span>)</code></pre></div>
<blockquote>
<p>more information, see ?plot3D_lpc</p>
</blockquote>
<ul>
<li>res.lpc. data after run get_biopath_score</li>
<li>normals. a vector of the samples, if samples is Normal then TRUE, else Tumor then FALSE</li>
<li>outputHTML. if need output this plot as a html file is TRUE, if not then FALSE</li>
<li>outputDir. if outputHTML=T, this is availiable, the output file directory</li>
<li>fileName. if outputHTML=T, this is availiable, the output file name.</li>
</ul>
</div>
</div>
<div id="examples" class="section level2">
<h2>Examples</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(biopathscore)
<span class="kw">data</span>(<span class="st">&quot;dat_brca&quot;</span>)
<span class="kw">data</span>(<span class="st">'v_brca_normals'</span>)
<span class="kw">data</span>(<span class="st">'l_kegg_gs_min'</span>)


test_pathway =<span class="st"> </span>l_kegg_gs_min$KEGG_DNA_REPLICATION


### Examples 1: not use maximize_stability = F, (if run slow, set attempts lower number)
pds =<span class="st"> </span><span class="kw">get_biopath_score</span>(<span class="dt">data =</span> dat_brca, <span class="dt">biopath_genes =</span> test_pathway, <span class="dt">normals =</span> v_brca_normals,
                        <span class="dt">attempts =</span> <span class="dv">100</span>, <span class="dt">maximize_stability =</span> F)


<span class="kw">plot3D_lpc</span>(pds, v_brca_normals,<span class="dt">drawLine =</span> F, <span class="dt">outputHTML =</span> F )
<span class="kw">plot3D_lpc</span>(pds, v_brca_normals,<span class="dt">drawLine =</span> T, <span class="dt">outputHTML =</span> F )


### Examples 1: use maximize_stability = T, very slow
pds2 =<span class="st"> </span><span class="kw">get_biopath_score</span>(<span class="dt">data =</span> dat_brca, <span class="dt">biopath_genes =</span> test_pathway, <span class="dt">normals =</span> v_brca_normals,
                        <span class="dt">attempts =</span> <span class="dv">100</span>, <span class="dt">maximize_stability =</span> T)


<span class="kw">plot3D_lpc</span>(pds2, v_brca_normals,<span class="dt">drawLine =</span> F, <span class="dt">outputHTML =</span> F )
<span class="kw">plot3D_lpc</span>(pds2, v_brca_normals,<span class="dt">drawLine =</span> T, <span class="dt">outputHTML =</span> F )</code></pre></div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
